packages/remix-testing/create-remix-stub.tsx

```diff
@@ -62,245 +35,25 @@ type RemixStubOptions = {
-    routeData: initialLoaderData || {},
-    manifest,
-    routeModules: createRouteModules(routes),
-  };
-}
-
-function createManifest(routes: AgnosticDataRouteObject[]): AssetsManifest {
-  return {
-    routes: createRouteManifest(routes),
-    entry: { imports: [], module: "" },
-    url: "",
-    version: "",
-  };
-}
-
-function createRouteManifest(
-  routes: AgnosticDataRouteObject[],
-  manifest?: RouteManifest<EntryRoute>,
-  parentId?: string
-): RouteManifest<EntryRoute> {
-  return routes.reduce((manifest, route) => {
-    if (route.children) {
-      createRouteManifest(route.children, manifest, route.id);
-    }
-    manifest[route.id!] = convertToEntryRoute(route, parentId);
-    return manifest;
-  }, manifest || {});
-}
-
-function createRouteModules(
-  routes: AgnosticDataRouteObject[],
-  routeModules?: RouteModules
-): RouteModules {
-  return routes.reduce((modules, route) => {
-    if (route.children) {
-      createRouteModules(route.children, modules);
-    }
-
-    modules[route.id!] = {
-      CatchBoundary: undefined,
-      ErrorBoundary: undefined,
-      // @ts-expect-error - types are still `agnostic` here
-      default: () => route.element,
-      handle: route.handle,
-      links: undefined,
-      meta: undefined,
-      // TODO: Check with Logan on how to handle the update here
-      // @ts-expect-error
-      unstable_shouldReload: undefined,
-    };
-    return modules;
-  }, routeModules || {});
-}
-
-const originalFetch =
-  typeof global !== "undefined" ? global.fetch : window.fetch;
-
-function monkeyPatchFetch(
-  queryRoute: StaticHandler["queryRoute"],
-  dataRoutes: StaticHandler["dataRoutes"]
-) {
-  let fetchPatch = async (
-    input: RequestInfo | URL,
-    init: RequestInit = {}
-  ): Promise<Response> => {
-    let request = new Request(input, init);
-    let url = new URL(request.url);
-
-    // if we have matches, send the request to mock routes via @remix-run/router rather than the normal
-    // @remix-run/server-runtime so that stubs can also be used in browser environments.
-    let matches = matchRoutes(dataRoutes, url);
-    if (matches && matches.length > 0) {
-      let response = await queryRoute(request);
-
-      if (response instanceof Response) {
-        return response;
-      }
-
-      return json(response);
-    }
-
-    // if no matches, passthrough to the original fetch as mock routes couldn't handle the request.
-    return originalFetch(request, init);
-  };
-
-  globalThis.fetch = fetchPatch;
-}
-
-function convertToEntryRoute(
-  route: AgnosticDataRouteObject,
-  parentId?: string
-): EntryRoute {
-  return {
-    id: route.id!,
-    index: route.index,
-    caseSensitive: route.caseSensitive,
-    path: route.path,
-    parentId,
-    hasAction: !!route.action,
-    hasLoader: !!route.loader,
-    module: "",
-    hasCatchBoundary: false,
-    hasErrorBoundary: false,
+    return <RouterProvider router={routerRef.current} />;
```

Comment:
I think I was expecting we'd still have a `RemixContext.Provider` here - I think at the moment if you test a component like `<Link prefetch="render">` it'll break?  We should be able to create those from `router.routes` I think - similar to how we did it before

Reply:
oop, yep you're right